cores/arduino/Uart.cpp

```diff
@@ -102,11 +129,30 @@ int Uart::peek()
 
 int Uart::read()
 {
-  return rxBuffer.read_char();
+  int c = rxBuffer.read_char();
+
+  if (uc_pinRTS != NO_RTS_PIN) {
+    // if there is enough space in the RX buffer, assert RTS
+    if (rxBuffer.availableForStore() > RTS_RX_THRESHOLD) {
+      digitalWrite(uc_pinRTS, LOW);
+    }
+  }
+
+  return c;
 }
 
 size_t Uart::write(const uint8_t data)
 {
+  if (uc_pinRTS != NO_RTS_PIN) {
+    // assert RTS
+    digitalWrite(uc_pinRTS, LOW);
+  }
+
+  if (uc_pinCTS != NO_CTS_PIN) {
```

Comment:
mmmhhh there is no other way here? Ideally the CTS pin should mask the tx ISR, so you can always write in the buffer without blocking (until the buffer is full). What is not clear to me is if there is a way to trasparently re-enable the tx ISR.

Reply:
The latest commit addresses this.